.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,20 @@
+name: CodeQL
+on:
+  push: {}
+  schedule:
+    - cron: 0 0 * * *
```

Comment:
Why simply not on push?

Reply:
GitHub recommends you to run this check periodically so that new vulnerabilities can be identified even if no new code was pushed to the repository.